L1Trigger/TrackFindingTracklet/src/TrackletProcessor.cc

```diff
@@ -191,152 +204,58 @@ void TrackletProcessor::execute() {
-
-          int rzfinebinfirst = lookupbits.bits(0, 3);  //finephi
-          int next = lookupbits.bits(3, 1);            //next r/z bin
-          int start = lookupbits.bits(4, nbitsrzbin_);
-          int rzdiffmax = lookupbits.bits(lookupbits.nbits() - 3, 3);
-
-          if ((iSeed_ == 4 || iSeed_ == 5) && negdisk) {  //TODO - need to store negative disk
-            start += 4;
-          }
-          int last = start + next;
-
-          int nbins = 8;
-
-          unsigned int useregindex = (innerfinephi << innerbend.nbits()) + innerbend.value();
-          if (iSeed_ >= 4) {
-            //FIXME If the lookupbits were rationally organized this would be much simpler
-            int ir = ((start & 3) << 1) + (rzfinebinfirst >> 2);
-            useregindex = (useregindex << 3) + ir;
-          }
-
-          assert(useregindex < useregion_.size());
-          vector<bool> usereg = useregion_[useregindex];
-
-          TEData tedata;
-          tedata.stub_ = stub;
-          tedata.rzbinfirst_ = rzfinebinfirst;
-          tedata.start_ = start;
-          tedata.innerfinephi_ = innerfinephi;
-          tedata.rzdiffmax_ = rzdiffmax;
-          tedata.innerbend_ = innerbend;
-
-          for (unsigned int ireg = 0; ireg < settings_.nvmte(1, iSeed_); ireg++) {
-            assert(ireg < usereg.size());
-            if (!usereg[ireg])
-              continue;
-
-            for (int ibin = start; ibin <= last; ibin++) {
-              if (settings_.debugTracklet()) {
-                edm::LogVerbatim("Tracklet") << getName() << " looking for matching stub in bin " << ibin << " with "
-                                             << outervmstubs_->nVMStubsBinned(ireg * nbins + ibin) << " stubs";
-              }
-              assert(ireg * nbins + ibin < outervmstubs_->nBin());
-              int nstubs = outervmstubs_->nVMStubsBinned(ireg * nbins + ibin);
-              if (nstubs > 0) {
-                tedata.regions_.emplace_back(tuple<int, int, int>(ibin - start, ireg, nstubs));
-                countteall += nstubs;
-              }
-            }
-          }
-
-          if (!tedata.regions_.empty()) {
-            ntedata++;
-            //cout << getName() << " adding TE data" << endl;
-            tedatabuffer.store(tedata);
-          }
-        }
-        istub++;
-        if (istub >= innerallstubs_[imem]->nStubs()) {
-          istub = 0;
-          imem++;
-        }
+  TEData tedata;
+  TEData tedata__;
+  TEData tedata___;
+  bool goodtedata = false;
+  bool goodtedata__ = false;
+  bool goodtedata___ = false;
+
+  bool tebuffernearfull;
+  
+  for (unsigned int istep = 0; istep < maxStep_; istep++) {
+    if (print) {
+      CircularBuffer<TEData>& tedatabuffer = std::get<0>(tebuffer_);
+      unsigned int& istub = std::get<1>(tebuffer_);
+      unsigned int& imem = std::get<2>(tebuffer_);
+      cout << "istep=" << istep << " TEBuffer: " << istub << " " << imem << " " << tedatabuffer.rptr() << " "
```

Comment:
cout again

Reply:
This print statement is carefully crafted to match exactly a print statement in the HLS code. It has been, and likely will continue to be very helpful when tracking down differences between HLS and the emulations. I would like to leave this as is - changing to message logger will add additional printout.  We will probably have a few similar cases. What is the best way of dealing with this?